### PR TITLE
Tempfiles for data.tar.gz and control.tar.gz

### PR DESCRIPTION
**Pull request by xor-gate**
Initial work on tempfiles for data.tar.gz and control.tar.gz when packaging exceed RAM size.

### CODE REVIEW

**Comment by xor-gate** [comment]
ping @rikvdh

File: debpkg.go [hunk 1]
```diff
@@ -22,43 +22,58 @@ import (
...
+	data         data
+	digest       digest
+}
+
+var debpkgTempDir = "" // default temporary directory is os.TempDir
+
+// SetTempDir sets the temporary directory for intermediate files. When the directory doesn't
+//  exist it is automaticly created. The caller is responsible for the cleanup of tempdir.
+func SetTempDir(dir string) error {
+	if dir == "" {
```
Comment (rikvdh) [inline debpkg.go]: When the directory is emtpy you take the empty-directory as TempDir? Don't you want to get the system-wide temp-dir with `os.TempDir()`?

File: control.go [hunk 2]
```diff
@@ -203,39 +202,50 @@ func (deb *DebPkg) AddConffile(filename string) {
...
 
 	o += fmt.Sprintf("Package: %s\n", c.info.name)
-	o += fmt.Sprintf("Version: %s\n", c.info.version.full)
+	o += fmt.Sprintf("Version: %s\n", c.version())
 	o += fmt.Sprintf("Architecture: %s\n", c.info.architecture)
 	o += fmt.Sprintf("Maintainer: %s <%s>\n",
 		c.info.maintainer,
 		c.info.maintainerEmail)
-	o += fmt.Sprintf("Installed-Size: %d\n", installedSize)
+	o += fmt.Sprintf("Installed-Size: %d\n", uint64(math.Floor((float64(installedSize)/1024)+0.5)))
```
Comment (rikvdh) [inline control.go]: Just use `math.Ceil()` without the `floor(x+0.5)` which is also wrong.

File: debpkg.go (hunk 1 above)
Comment (xor-gate) [inline debpkg.go]: empty Directory is automatic os.TempDir, see os.TempFile

**Pull request merged**